### PR TITLE
php.packages.deployer: install shell completions

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
I thought shell completions would be helpful.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124817` at 7fc20593 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>php73Packages.deployer</li>
    <li>php74Packages.deployer</li>
    <li>php80Packages.deployer</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/php-packages/deployer/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/php-packages/deployer/default.nix [hunk 1]
```diff
@@ -11,12 +11,17 @@ mkDerivation rec {
...
 
   installPhase = ''
     mkdir -p $out/bin
     install -D $src $out/libexec/deployer/deployer.phar
     makeWrapper ${php}/bin/php $out/bin/dep --add-flags "$out/libexec/deployer/deployer.phar"
+
+    # fish support currently broken: https://github.com/deployphp/deployer/issues/2527
+    installShellCompletion --cmd dep \
+      --bash <($out/bin/dep autocomplete --install) \
+      --zsh <($out/bin/dep autocomplete --install)
```
Comment (SuperSandro2000) [inline pkgs/development/php-packages/deployer/default.nix:24]: Are those commands supposed to be the same?

Comment (aanderse) [reply pkgs/development/php-packages/deployer/default.nix:24]: Yes. This is listed in documentation... even for `fish` completion. Which explains why `fish` completion doesn't work :disappointed: I only tested the `bash` completion. I have never really dabbled around with `zsh` to know if it is realistic to expect this to actually work or not...

Comment (SuperSandro2000) [reply pkgs/development/php-packages/deployer/default.nix:24]: bash scripts mostly work under bash so I just trust upstream that it works for now.

**Pull request merged**